### ISSUE TITLE
JSON_INVALID lint error returns null file name instead of actual JSON file name

### ISSUE THREAD

**Issue by pdehaan** [mozilla/addons-linter#3282]
Via https://github.com/mozilla/web-ext/issues/1993

---

### Is this a feature request or a bug?
Bug?

### What is the current behavior?

I was trying to test the webExt API to lint a project and noticed that it seems to send the `webExt.cmd.lint` output directly to STDOUT instead of returning it to my code, so I piped the output to disk (res.json) and noticed that I get an `JSON_INVALID` error because the file is temporarily empty (and in my `sourceDir`, but the error gives me a `null` value for the `file` prop (and I feel like it should know the file path, and lists it in the `emptyFiles[]` array in the `metadata`):

```json
{
    "count": 2,
    "summary": {
        "errors": 2,
        "notices": 0,
        "warnings": 0
    },
    "metadata": {
        "totalScannedFileSize": 233371,
        "emptyFiles": [
            "res.json"
        ],
        "jsLibs": {},
        "unknownMinifiedFiles": []
    },
    "errors": [
        {
            "_type": "error",
            "code": "TYPE_NO_MANIFEST_JSON",
            "message": "manifest.json was not found",
            "description": "No manifest.json was found at the root of the extension. The package file must be a ZIP of the extension's files themselves, not of the containing directory. See: https://mzl.la/2r2McKv for more on packaging."
        },
        {
            "_type": "error",
            "code": "JSON_INVALID",
            "message": "Your JSON is not valid.",
            "description": "Unexpected end of JSON input",
            "file": null
        }
    ],
    "notices": [],
    "warnings": []
}
```

```js
// lint-test.js
const webExt = require("web-ext");

webExt.cmd.lint({
    sourceDir: ".",
    output: "json",
    verbose: false,
    pretty: true,
  });
```

### What is the expected or desired behavior?

If I get a `JSON_INVALID` error, it should list the correct filename instead of `null`.

### Version information (for bug reports)

* **Firefox version**:
* **Your OS and version**: macOS 10.15.6
* Paste the output of these commands:

```sh
> node --version && npm --version && web-ext --version

v14.7.0
6.14.7
5.0.0
```

**Comment by pdehaan** [comment]
Here's a pure **addons-linter** version:

```js
const { createInstance } = require("addons-linter");

const VERBOSE = Boolean(process.env.VERBOSE);

const linter = createInstance({
  config: {
    _: [process.cwd()],
    boring: false,
    logLevel: VERBOSE ? "debug" : "fatal",
    metadata: false,
    output: "none",
    pretty: true,
    runAsBinary: false,
    selfHosted: false,
    stack: VERBOSE,
    warningsAsErrors: false,

    shouldScanFile(fileName) {
      if (fileName.startsWith("node_modules")) {
        return false;
      }
      console.info(`Scanning "${fileName}"`);
      return true;
    },
  },
});

main();

async function main() {
  try {
    const res = await linter.run();
    console.log(JSON.stringify(res, null, 2));
  } catch (err) {
    console.error("addons-linter failure: ", err);
  }
}
```

### Output

```
Scanning ""
Scanning "empty.json"
Scanning "index.js"
Scanning "package-lock.json"
Scanning "package.json"

{
  "count": 2,
  "summary": {
    "errors": 2,
    "notices": 0,
    "warnings": 0
  },
  "metadata": {
    "totalScannedFileSize": 98820,
    "emptyFiles": [
      "empty.json"
    ],
    "jsLibs": {},
    "unknownMinifiedFiles": []
  },
  "errors": [
    {
      "_type": "error",
      "code": "TYPE_NO_MANIFEST_JSON",
      "message": "manifest.json was not found",
      "description": "No manifest.json was found at the root of the extension. The package file must be a ZIP of the extension's files themselves, not of the containing directory. See: https://mzl.la/2r2McKv for more on packaging."
    },
    {
      "_type": "error",
      "code": "JSON_INVALID",
      "message": "Your JSON is not valid.",
      "description": "Unexpected end of JSON input",
      "file": null
    }
  ],
  "notices": [],
  "warnings": []
}
```

**Comment by regseb** [comment]
I think it's commit https://github.com/mozilla/addons-linter/commit/087cac8e5979f17eb55979d2aa5ea6c046fce261 that introduced the problem (find with _git-bisect_).

**Comment by AlexandraMoga** [comment]
JSON files that triggered a validation error are now returned in the linter results:

[before fix]
![image](https://user-images.githubusercontent.com/31961530/97441516-7ca1c300-1931-11eb-9f5e-9a8fd93531fc.png)

[after fix]
![image](https://user-images.githubusercontent.com/31961530/97441563-89261b80-1931-11eb-8b05-0f7ab51540c6.png)

[before fix]
![image](https://user-images.githubusercontent.com/31961530/97441665-a2c76300-1931-11eb-9957-53eafeaf2aff.png)

[after fix]
![image](https://user-images.githubusercontent.com/31961530/97441629-9a6f2800-1931-11eb-81f4-96bf35873e15.png)
